### ISSUE TITLE
Parsing value with click event 

### ISSUE THREAD

**Issue by PhilipRying** [msaraiva/surface#96]
```
defmodule Panel do
  use Surface.LiveComponent
  alias SurfaceBulma.Button
  alias SurfaceBulma.Table
  alias SurfaceBulma.Table.Column
  alias Admin.Shed

  data(drawers, :map)

  def mount(socket) do
    if connected?(socket), do: Admin.Shed.subscribe()
    {:ok, fetch(socket)}

    # assign(socket, drawers: Admin.Shed.list_drawers()
  end

  defp fetch(socket) do
    assign(socket, drawers: Admin.Shed.list_drawers())
  end

  def render(assigns) do
    ~H"""
    <Table data={{ {number,drawer} <- @drawers }} bordered>
    <Column label="#">
    {{ number }}
    </Column>
    <Column label="Status">
      {{ Kernel.inspect(drawer.status) }}
    </Column>
    <Column label="Code">
    {{ drawer.code }}
    </Column>
    <Column label="send code">
    <button :on-phx-click="send_code" :on-phx-value-drawer={{ number }}> Send Code </button>
    </Column>
    </Table>

    <button :on-phx-click="add" > Update drawer </button>

    """
  end

  def handle_event("add", _value, socket) do
    Admin.Shed.update_drawer(2, Admin.Drawer.new(:in_use, 888_888))
    # drawers = Admin.Shed.list_drawers()

    # drawers = Map.put(drawers, 2, Admin.Drawer.new(:in_use, 888_888))
    # socket = assign(socket, :drawers, drawers)
    {:noreply, socket}
  end

  def handle_event("send_code", %{"drawer" => id}, socket) do
    # Add the send code function use id to know which drawer.
    {:noreply, socket}
  end

  def fetch_drawers(panel_id) do
    send_update(__MODULE__, id: panel_id, drawers: Admin.Shed.list_drawers())
  end
end
```

I am trying to parse the drawer id from a table in a button, but this doesn't work. How can i fix this?

**Comment by msaraiva** [comment]
Hi @PhilipRying!

There's no `:on-phx-value-drawer` directive. The `:on-*` variants are only for events. You should use phoenix's built-in `phx-value-*` instead:

```jsx
<button :on-phx-click="send_code" phx-value-drawer={{ number }}> Send Code </button>
```

For a list of all available directives, see http://surface-demo.msaraiva.io/components_basics.

For details on phoenix's `phx-value-*`, see https://hexdocs.pm/phoenix_live_view/bindings.html#click-events.

Please, let me know if this fixes the issue.

**Comment by PhilipRying** [comment]
Thank you so much! it works.🙂

**Comment by arathunku** [comment]
Sorry for bumping old thread (I can open a new one but it might be more appropriate for discussions than issues...)


What would be the easiest way to capture `phx-value-*` and forward it in components? Let's say I've got a button, and it supports `click` prop, passing it to `:on-click`, but I'd like open a possibility of supplementing it with `phx-values`. I can see in `Link` there are a lot of helpers to do that, but they're not exposed anywhere :( I've also checked https://github.com/surface-ui/surface_bulma, but it looks like it has the same problem?  

I'm thinking about something like this:

```jsx
<Button click="delete_identity" phx-value-foobar={{}} ...or more>
  Delete
</Button>
```

The alternative could be `click={{ {:delete_identity, [foobar: ...] }}` and handle it inside the component but it's less clear :(
